src/Functions/array/arrayCompact.cpp

```diff
@@ -60,23 +57,22 @@ struct ArrayCompactImpl
             if (src_pos < src_offset)
             {
                 /// Insert first element unconditionally.
-                res_values[res_pos] = src_values[src_pos];
+                res_values_column->insertFrom(src_values, src_pos);
 
                 /// For the rest of elements, insert if the element is different from the previous.
                 ++src_pos;
                 ++res_pos;
                 for (; src_pos < src_offset; ++src_pos)
                 {
-                    if (!bitEquals(src_values[src_pos], src_values[src_pos - 1]))
+                    if (!bitEquals(check_values[src_pos], check_values[src_pos - 1]))
                     {
-                        res_values[res_pos] = src_values[src_pos];
+                        res_values_column->insertFrom(src_values, src_pos);
```

Comment:
I think, it's worth to make  a specialization for `Array` of integers and operate over arrays as it was before and avoid virtual call in a loop in that case. It makes sense, because `arrayCompact` with single `Array` of integers has been slowed down.

`clickhouse benchmark <<< "select arrayCompact(range(number % 1000)) from numbers(100000) format Null" --cumulative`

In your PR:
```
localhost:9000, queries 62, QPS: 4.360, RPS: 571169.982, MiB/s: 4.358, result RPS: 0.000, result MiB/s: 0.000.

0.000%		0.204 sec.	
10.000%		0.212 sec.	
20.000%		0.215 sec.	
30.000%		0.217 sec.	
40.000%		0.219 sec.	
50.000%		0.222 sec.	
60.000%		0.227 sec.	
70.000%		0.239 sec.	
80.000%		0.245 sec.	
90.000%		0.253 sec.	
95.000%		0.261 sec.	
99.000%		0.281 sec.	
99.900%		0.301 sec.	
99.990%		0.301 sec.
```

In master:

```
localhost:9000, queries 133, QPS: 13.072, RPS: 1712517.749, MiB/s: 13.065, result RPS: 0.000, result MiB/s: 0.000.

0.000%		0.051 sec.	
10.000%		0.053 sec.	
20.000%		0.054 sec.	
30.000%		0.054 sec.	
40.000%		0.055 sec.	
50.000%		0.056 sec.	
60.000%		0.057 sec.	
70.000%		0.060 sec.	
80.000%		0.135 sec.	
90.000%		0.138 sec.	
95.000%		0.140 sec.	
99.000%		0.143 sec.	
99.900%		0.155 sec.	
99.990%		0.155 sec.
```